packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -51,12 +62,24 @@ function build(syscall, _state, makeRoot, forVatID) {
 
   const outstandingProxies = new WeakSet();
 
+  /** Map in-vat object references -> vat slot strings.  Uses a weak map so
+      that vat objects can be GC'd. */
```

Comment:
.. however they currently can't because `slotToVal` keeps local objects alive. One task before us is to figure out how to safely decref these from the kernel side. Another task is to safely (deterministically) learn when a local object has been GCed so we can tell the kernel to drop our reference to it (WeakRefs are the non-safe part of the answer)

Reply:
Well OK then